### DISCUSSION TITLE
E3V2-BLTouch-3x3-v4.2.7-v1.3.4 bin using z endstop for bltouch? SOLVED

### DISCUSSION THREAD

**Discussion by QuinnWede** [Jyers/Marlin#1064] [Q&A] [answered]
Does the "E3V2-BLTouch-3x3-v4.2.7-v1.3.4" bin have "#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN" Uncommented? My bltouch is not working and that seems like a possible reason.

Edit: My problem wasn't the firmware, the board's pinout was labeled completely backwards

**Comment by smooreace** [comment, marked as answer]
Not likely. The "proper" way is to NOT use the z limit circuit. But, you
can look at the config file used if you download the source files. Which
you will also have to do to compile anything other than what is offered.

**Reply by Nwjx** [reply]
I too have the same problem. 

I've come to the conclusion that the precompiled files only work with bltouch that comes with the other type of cable. The cable that came with mine uses an adaptor board that plugs into the screen connector as well.

Installation guide for the reference bltouch;
https://www.makersupplies.sg/blogs/tutorials/how-to-install-bl-touch-on-the-ender-3-pro-v2-3d-printer-with-adapter-board-version

Edit:
Scouring the discussions and confirming it myself. you will have to get your hands a bit dirty by manually changing the pins and squeezing the connectors into the dedicated port.
With reference to smith3d's guide, I'm using the third party extension cable although the cable colours are different, using the guide to switch out the wires to fit the order as shown finally enabled the use of the dedicated port. 
Try not to be confused with the colours.

Original thread:
https://github.com/Jyers/Marlin/discussions/800
source:
https://smith3d.com/ender-3-v2-bltouch-firmware-installation-guide-by-smith3d-com/

**Reply by QuinnWede** [reply]
This is the setup I have right now:
https://github.com/Jyers/Marlin/discussions/1054#discussion-3427285

**Reply by Nwjx** [reply]
Comparing your setup to smith3d's antclabs setup. doesn't look like there should be an issue caused by the connection. You may want to ensure that you have properly flashed your printer. Right now, that is the only thing I could think of.
